### PR TITLE
issue 402.1: share sync thread between IResearchView and IResearchViewDBServer

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

**Comment by vasiliy-arangodb** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/147/

File: arangod/Cluster/ClusterInfo.cpp [hunk 1]
```diff
@@ -1113,6 +1113,64 @@ std::shared_ptr<LogicalView> ClusterInfo::getView(
...
+
+      if (itr != views.end()) {
+        return itr->second;
+      }
+    }
+
+    return nullptr;
+  };
+
+  if (std::this_thread::get_id() == _planLoader) {
```
Comment (graetzer) [inline arangod/Cluster/ClusterInfo.cpp]: is this really a good idea ? Does the validity of this ID depend on the way the thread was created  (i.e. are we using `std::thread` everywhere now )?

Comment (vasiliy-arangodb) [reply arangod/Cluster/ClusterInfo.cpp]: AFAIK std::this_thread::get_id() doesn't depend on the usage of std::thread

Comment (ObiWahn) [reply arangod/Cluster/ClusterInfo.cpp]: There is no guarantee that it must work with non std::threads. But as the std::thread stuff is just a thin wrapper around the native stuff it is very likely to work.

Comment (ObiWahn) [reply arangod/Cluster/ClusterInfo.cpp]: http://eel.is/c++draft/thread.threads#thread.thread.id-1

**Pull request merged**